### PR TITLE
Some code refactor on the Raft layer

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6233

Problem Summary:

### What is changed and how it works?

* Move the debugging method `KVStore::handleWriteRaftCmd` with `raft_cmdpb::RaftCmdRequest` to `RegionBench ::applyWriteRaftCmd` to reduce code complexity on `KVStore`
* Use early exit in some if-branches
* Rename `gcRegionPersistedCache` to `gcPersistedRegion`
* Refine some logging about keyspace and table_id

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 1]
```diff
@@ -232,52 +233,11 @@ RegionManager::RegionWriteLock KVStore::genRegionMgrWriteLock(const KVStoreTaskL
 }
 
 EngineStoreApplyRes KVStore::handleWriteRaftCmd(
-    raft_cmdpb::RaftCmdRequest && request,
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Transaction/KVStore.cpp:235]: The debugging method with `raft_cmdpb::RaftCmdRequest` is moved to `dbgTools.cpp`

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**